assets/js/wc-gateway-ppec-generate-cart.js

```diff
@@ -14,20 +14,29 @@
 
 	$( '#woo_pp_ec_button_product' )
 		.on( 'enable', function() {
-			$( '#woo_pp_ec_button_product' ).css( {
-				'cursor': '',
-				'-webkit-filter': '', // Safari 6.0 - 9.0
-				'filter': '',
-			} );
-			$( '#woo_pp_ec_button_product > *' ).css( 'pointer-events', '' );
+			$( '#woo_pp_ec_button_product' )
+				.css( {
+					'cursor': '',
+					'-webkit-filter': '', // Safari 6.0 - 9.0
+					'filter': '',
+				} )
+				.off( 'mouseup' )
+				.find( '> *' )
+				.css( 'pointer-events', '' );
 		} )
 		.on( 'disable', function() {
-			$( '#woo_pp_ec_button_product' ).css( {
-				'cursor': 'not-allowed',
-				'-webkit-filter': 'grayscale( 100% )', // Safari 6.0 - 9.0
-				'filter': 'grayscale( 100% )',
-			} );
-			$( '#woo_pp_ec_button_product > *' ).css( 'pointer-events', 'none' );
+			$( '#woo_pp_ec_button_product' )
+				.css( {
+					'cursor': ( variation_valid ? 'pointer' : 'not-allowed' ),
```

Comment:
Hey @chickenn00dle this PR is awesome 💯 
I tested this on variable products and it's looking much better now. Woo!

One question I had was with this `variation_valid` condition being used to display either a pointer or the not-allowed cursor.

What was the reasoning for this?

When I was testing other 3rd party products like a Mix and Match product, I noticed the cursor has changed going from master to this branch. Here's a video starting on master: https://d.pr/v/ICC8tc

It's not a big issue but I'm curious to understand why this was needed and if I'm missing something 🤔 ? Was there a need for showing a pointer cursor when the `variation_valid` is set to true but the button is still disabled?

Reply:
Thanks @mattallan!

> One question I had was with this variation_valid condition being used to display either a pointer or the not-allowed cursor.

I added this condition to mimic the Add to cart button in cases where product addons were being used:

![Screen Capture on 2020-05-07 at 10-14-35](https://user-images.githubusercontent.com/17905991/81244380-3b4cec80-904d-11ea-9778-35a32d48aa4d.gif)

I only tested with this extension as it was the one brought up in the original issue. That said, I'll see if I can find an alternative that works more universally.

Reply:
Hmm. It looks like Mix and Match Products is a paid extension, so I'm not able to test on my end. But my immediate thoughts are to default to `not-allowed` here unless both `variation_valid` and `fields_valid` are true:

`'cursor': ( fields_valid && variation_valid ? 'pointer' : 'not-allowed' ),`
 
This would result in a pointer mismatch in the case of Product Addons but would still make sense for the user. It would also remove the inconsistent pointer mismatch you pointed out with Mix and Match Products (I'm assuming).

What are your thoughts?

Reply:
Thanks for sharing this @chickenn00dle I wasn't aware there was a different in pointers for Mix and Match and Product Add-ons.

This is a pretty minor change overall but I think your latest approach sounds good to me:

```
'cursor': ( fields_valid && variation_valid ? 'pointer' : 'not-allowed' ),
```

Reply:
Great! 

I made changes in https://github.com/woocommerce/woocommerce-gateway-paypal-express-checkout/pull/707/commits/909b3f04bed5421de69be9d86acdd428808ac228

I realized if both `fields_value` and `variation_valid` are true, the buttons will be enabled. So I removed the conditional check altogether.